src/ReactHardwareComponent.js

```diff
@@ -145,48 +163,34 @@ ReactHardwareComponent.Mixin = {
     context: Object // secret context, shhhh
   ) {
     rootID = typeof rootID === 'object' ? rootID._rootNodeID : rootID;
-
     this._rootNodeID = rootID;
-    if (this._currentElement.props[CONTAINER_KEY] === CONTAINER_VALUE) {
-      this.initializeChildren(
-        this._currentElement.props.children,
-        transaction,
-        context
-      );
-
-      return rootID;
-    }
 
-    const payload = create(
-      this._currentElement.props, // next props
-      this.viewConfig.validAttributes
-    );
-
-    if (payload) {
-      if (process.env.NODE_ENV !== 'production') {
-        HardwareManager.validatePayloadForPin(rootID, payload);
+    switch (this._currentElement.type) {
+      case 'pin': {
+        const payload = create(
+          this._currentElement.props, // next props
+          this.viewConfig.validAttributes
+        );
+        if (payload) {
+          if (process.env.NODE_ENV !== 'production') {
+            HardwareManager.validatePayloadForPin(rootID, payload);
+          }
+        }
+        HardwareManager.setPayloadForPin(rootID, payload);
+        // TODO register listeners
+        // TODO: figure out what this should be for RH.
+        // ReactNative initializeChildren(children, tag, transaction, context);
+        // https://github.com/facebook/react-native/blob/9f48c004ba866aa24d17242a817929462a091179/Libraries/ReactNative/ReactNativeBaseComponent.js
+        //
+        // ReactDOM _createInitialChildren(transaction, props, context, lazyTree);
+        //   https://github.com/facebook/react/blob/master/src/renderers/dom/shared/ReactDOMComponent.js#L725
```

Comment:
while cleaning up this PR this entire comment block could actually be removed :smile: